### DISCUSSION TITLE
Locations on a map

### DISCUSSION THREAD

**Discussion by BWibo** [FraunhoferIOSB/FROST-Server#536] [Show and tell]
Hey there, 

I've been putting together some very basic examples on how to query and draw `Locations` (-> Markers ) and `HistoricalLocations`  (-> Trajectory) on a [Leaflet.js](https://leafletjs.com) map. This could possibly be a cool feature for the FROST-Server landing page or be useful for you guys in some other way.

https://github.com/tum-gis/FROST-on-a-map

Here is an exmple: https://www.3dcitydb.org/3dcitydb/fileadmin/public/frost-on-a-map-img/

**Comment by hylkevds** [comment]
That's really cool. We've not done much with moving sensors ourselves, but it's cool to see some use cases!

Are you aware of the project STAM: https://github.com/DataCoveEU/STAM/
That mapper doesn't deal with HistoricalLocations, but does do staged-loading, which becomes relevant when there are thousands of Things. (Demo: https://datacoveeu.github.io/API4INSPIRE/maps/AirQuality.html)
It may be cool to combine forces :)

I did notice the URL of your server is not quite conform the standard, since you leave out the version number... The version number must be exposed in the URL, to ensure the client knows which version of the standard is used.

**Comment by BWibo** [comment]
Hey there,
thanks for the for the hint to STAM and regarding the URL of the service. Currently, we are in the stage of doing or first steps with moving sensors. We are hosting some student projects heading in this direction in the upcoming months. I'll let you know when we get presentable results.
Staged-loading is interesting for us as well. We are currently building up sensor services based on LoRaWAN and SensorThingsAPI for the [World Agricultural Systems Center of TU Munich](https://www.hef.tum.de/startseite/) and I believe we can expect larger clusters of sensor things there in the upcoming years. 

@HEFLoRa: 
* The [STAM project](https://github.com/DataCoveEU/STAM/) could be something for you, right?
* We should fix the URL of the HEF FROST instance, see above.

**Comment by KathiSchleidt** [comment]
Not quite relevant, but too cool not to mention if you're playing with multiple Locations per Things - I much like Hylke's "misuse" of this functionality to provide different resolutions of the same Thingy - done that for European NUTS regions and Demography Data from EuroStat.
Here the STA: https://demography.k8s.ilt-dmz.iosb.fraunhofer.de/v1.1
Here the STAM viewer that takes the scales into account when zooming (cool are the Fjords of Norway, zoom in and watch them get detailed): https://datacoveeu.github.io/API4INSPIRE/maps/Demography.html

**Comment by ksonda** [comment]
We're doing something similar to STAM here, https://github.com/internetofwater/sta-dashboard  eventually to be packaged in docker-compose with FROST but also standalone. Demo here https://sta-client.internetofwater.dev
